src/TLuaInterpreterMapper.cpp

```diff
@@ -579,7 +579,12 @@ int TLuaInterpreter::addRoom(lua_State* L)
     const bool added = host.mpMap->addRoom(id);
     lua_pushboolean(L, added);
     if (added) {
-        host.mpMap->setRoomArea(id, -1, false);
+        int areaID = -1;
+        if (lua_gettop(L) >= 2) {
+            areaID = getVerifiedInt(L, __func__, 2, "areaID");
+        }
+        // defer area calculations as all new rooms are initialised at 0,0,0 anyway
```

Comment:
🤔 If we are extending room creation to put them in the wanted area directly why not go the whole hog and accept the coordinates to position them in the area as well...? Fair enough - I am not sure what the Lua API would look like for *that*!

Reply:
Adding room coordinates isn't slow, so leaving it be for now